src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java

```diff
@@ -1505,6 +1515,170 @@ public void testGenerateXpathSuppressionOptionCustomTabWidth(@SysErr Capturable
+        final String expected = "";
+
+        assertMainReturnCode(0, "-c", getPath("InputMainConfig-empty.xml"),
+                "--generate-suppression", getPath("InputMainComplexityOverflow.java"));
+        assertWithMessage("Unexpected output log")
+            .that(systemOut.getCapturedData())
+            .isEqualTo(expected);
+        assertWithMessage("Unexpected system error log")
+            .that(systemErr.getCapturedData())
+            .isEqualTo("");
+    }
+
+    @Test
+    public void testGenerateSuppressionOptionCustomOutput(@SysErr Capturable systemErr)
+            throws IOException {
+        final String expected = addEndOfLine(
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>",
+                "<!DOCTYPE suppressions PUBLIC",
+                "    \"-//Checkstyle//DTD Checkstyle Configuration 1.3"
+                    + "//EN\"",
+                "    \"https://checkstyle.org/dtds/configuration_1_3.dtd\">",
+                "<suppressions>",
+                "<suppress",
+                "       files=\"InputMainGenerateXpathSuppressionsTabWidth.java\"",
+                "       checks=\"ExplicitInitializationCheck\"",
+                "\"/>",
+                "</suppressions>");
+        final File file = new File(temporaryFolder, "file.output");
+        assertMainReturnCode(0, "-c", getPath("InputMainConfig-xpath-suppressions.xml"), "-o",
+                file.getPath(), "--generate-suppression",
+                getPath("InputMainGenerateXpathSuppressionsTabWidth.java"));
+        try (BufferedReader br = Files.newBufferedReader(file.toPath())) {
+            final String fileContent = br.lines().collect(Collectors.joining(EOL, "", EOL));
+            assertWithMessage("Unexpected output log")
+                .that(fileContent)
+                .isEqualTo(expected);
+            assertWithMessage("Unexpected system error log")
+                .that(systemErr.getCapturedData())
+                .isEqualTo("");
+        }
+    }
+
+    @Test
+    public void testGenerateSuppressionOptionDefaultTabWidth(@SysErr Capturable systemErr,
+            @SysOut Capturable systemOut) {
+        final String expected = addEndOfLine(
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>",
+                "<!DOCTYPE suppressions PUBLIC",
+                "    \"-//Checkstyle//DTD Checkstyle Configuration 1.3"
+                    + "//EN\"",
+                "    \"https://checkstyle.org/dtds/configuration_1_3.dtd\">",
+                "<suppressions>",
+                "<suppress",
+                "       files=\"InputMainGenerateXpathSuppressionsTabWidth.java\"",
+                "       checks=\"ExplicitInitializationCheck\"",
+                "\"/>",
+                "</suppressions>");
+
+        assertMainReturnCode(0, "-c", getPath("InputMainConfig-xpath-suppressions.xml"),
+                "--generate-suppression",
+                getPath("InputMainGenerateXpathSuppressionsTabWidth.java"));
```

Comment:
please do copy of Input file and name it wihtout Xpath, We try to not reuse Inputs between features, as it very painful to in maintenance later on.
Separate Inputs are good, they can serve exact goal of feature and easily to be updated without affecting others.